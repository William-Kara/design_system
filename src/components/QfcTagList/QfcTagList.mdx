import { ArgTypes, Meta } from '@storybook/blocks';
import QfcTagList from '@/components/QfcTagList/QfcTagList.vue';

<Meta title="Modules/TagList" />

# Definition & Usage

## Props

The following props are available for the QfcTagList component:

<ArgTypes of={QfcTagList} />

# Usage

To use QfcTagList, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/bulma/elements/tag.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcTag.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcTagList.scss';

```

```html
<template>
  <QfcTagList :tagList="['tag1', 'tag2', 'tag3']" />
</template>
```
