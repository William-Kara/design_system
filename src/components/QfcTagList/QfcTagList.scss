.qfc-tag__list {
  span {
    margin-bottom: $q-generic-space-3;
  }
  span:not(:last-child) {
    margin-right: $q-generic-space-2;
  }
}
