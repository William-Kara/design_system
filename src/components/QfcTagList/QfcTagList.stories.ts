import TagList from './QfcTagList.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof TagList> = {
  component: TagList,
  title: 'Modules/TagList',
  argTypes: {
    tagList: {
      defaultValue: ['tag1', 'tag2', 'tag3'],
    },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    tagList: ['tag1', 'tag2', 'tag3'],
  },
};
