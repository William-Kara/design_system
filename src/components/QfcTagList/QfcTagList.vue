<template>
  <div class="qfc-tag__list">
    <QfcTag v-for="(tag, index) in tagList" :label="tag" :key="index" />
  </div>
</template>

<script lang="ts" setup>
import QfcTag from '@/components/QfcTag/QfcTag.vue';

type Props = {
  tagList: string[];
};
defineProps<Props>();
</script>
<style lang="scss" scoped src="./QfcTagList.scss" />
