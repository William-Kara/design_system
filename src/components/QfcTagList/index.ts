import QfcTagList from './QfcTagList.vue';

export { QfcTagList };
export default QfcTagList;
