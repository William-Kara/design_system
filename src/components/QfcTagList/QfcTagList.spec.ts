import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcTagList from './QfcTagList.vue';

let wrapper: any = null;

const propsValues = {
  tagList: ['tag1', 'tag2', 'tag3'],
};

beforeEach(() => {
  wrapper = shallowMount(QfcTagList, {
    propsData: propsValues,
  });
});

describe('QfcTagList Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
