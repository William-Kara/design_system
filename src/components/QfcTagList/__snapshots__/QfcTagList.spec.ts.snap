// Vitest Snapshot v1

exports[`QfcTagList Component > should render correctly 1`] = `
"<div data-v-a9123f78=\\"\\" class=\\"qfc-tag__list\\">
  <qfc-tag-stub data-v-a9123f78=\\"\\" label=\\"tag1\\"></qfc-tag-stub>
  <qfc-tag-stub data-v-a9123f78=\\"\\" label=\\"tag2\\"></qfc-tag-stub>
  <qfc-tag-stub data-v-a9123f78=\\"\\" label=\\"tag3\\"></qfc-tag-stub>
</div>"
`;
